compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -1247,8 +1243,8 @@ impl<'a> Parser<'a> {
 
         let lazy_impl = LazyTokenStreamImpl {
             start_token,
+            num_calls: self.token_cursor.num_next_calls - cursor_snapshot.num_next_calls,
             cursor_snapshot,
-            num_calls,
             desugar_doc_comments: self.desugar_doc_comments,
         };
         Ok((ret, Some(LazyTokenStream::new(lazy_impl))))
```

Comment:
Do you still want `collect_tokens` to return an `Option<LazyTokenStream>`?

Reply:
I want to include the
```
let (thing, tokens) = self.collect_tokens(...);
if thing.tokens.is_none() {
    thing.tokens = tokens;
}
```
logic into `collect_tokens` itself instead, but that's probably a task for another PR.